lib/fs.js

```diff
@@ -683,26 +687,28 @@ ObjectDefineProperty(read, internalUtil.customPromisifyArgs,
  *   offset?: number;
  *   length?: number;
  *   position?: number | bigint;
- *   }} [offset]
+ *   }} [offsetOrOptions]
  * @returns {number}
  */
-function readSync(fd, buffer, offset, length, position) {
+function readSync(fd, buffer, offsetOrOptions, length, position) {
   fd = getValidatedFd(fd);
 
   validateBuffer(buffer);
 
-  if (arguments.length <= 3) {
-    // Assume fs.readSync(fd, buffer, options)
-    const options = offset || ObjectCreate(null);
+  let offset = offsetOrOptions;
+  if (arguments.length <= 3 || typeof offsetOrOptions === 'object') {
+    if (offsetOrOptions !== undefined) {
+      validateObject(offsetOrOptions, 'options', { nullable: true });
+    }
 
     ({
       offset = 0,
       length = buffer.byteLength - offset,
       position = null
-    } = options);
+    } = offsetOrOptions ?? ObjectCreate(null));
   }
 
-  if (offset == null) {
+  if (offset === false || offset === undefined) {
```

Comment:
Since this is breaking anyway, let's not keep `false` as a valid value (unless it's too breaking maybe?)
```suggestion
  if (offset === undefined) {
```

Reply:
That's right, it should be invalidated. This check could be still a subject to change (due to https://github.com/nodejs/node/pull/42768#issuecomment-1102322560 if `null` should be allowed again), but looks correct rn.